controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1732,49 +1732,25 @@ private long getNewEpochMergeCount(final long mergeCountPrevEpoch, ImmutableList
      * This method will ignore any INVALID_TIME or INVALID_POSITION related failures in noting marks for writers.
      * This is because those will typically arise from idempotent commit case where this and a transaction with higher 
      * position and time may already have been committed and the overall mark for the writer may already have progressed.
-     * 
-     * @param committingTransactionsRecord Committing transaction record
+     *
      * @return A completableFuture, which when completed will have marks reported for all transactions in the committing 
      * transaction record for which a writer with time and position information is available. 
      */
-    CompletableFuture<Void> generateMarksForTransactions(CommittingTransactionsRecord committingTransactionsRecord,
-                                                         OperationContext context) {
+    CompletableFuture<Void> generateMarksForTransactions(OperationContext context,
+                                                         Map<String, CommitRequestHandler.TxnWriterMark> writerMarks) {
```

Comment:
should we move txn writer mark class out of commit request handler, given that it is to be used in the store's api. 
it should be in the store and used by commit request handler

Reply:
fixed